Content.Server/NPC/Systems/NPCSteeringSystem.Context.cs

```diff
@@ -322,19 +323,19 @@ private float MapValue(float value, float minValue, float maxValue)
                 !bodyQuery.TryGetComponent(ent, out var otherBody) ||
                 !otherBody.Hard ||
                 !otherBody.CanCollide ||
-                ((body.CollisionMask & otherBody.CollisionLayer) == 0x0 &&
-                (body.CollisionLayer & otherBody.CollisionMask) == 0x0))
+                (mask & otherBody.CollisionLayer) == 0x0 &&
+                (layer & otherBody.CollisionMask) == 0x0)
```

Comment:
I know other bodies can be non-hard but if we're really going to be doing it for all of them then I would rather just add additional fields with the cached values at that point.